### PR TITLE
Fix some misc. issues discovered in rc1

### PR DESCRIPTION
**Pull request by kzantow**
- [x] https://issues.jenkins-ci.org/browse/JENKINS-42788 - hide the edit link when no permissions
- [x] https://issues.jenkins-ci.org/browse/JENKINS-42735 - issue with branches containing slashes
- [x] https://issues.jenkins-ci.org/browse/JENKINS-42790 - remove the toast
- [x] https://issues.jenkins-ci.org/browse/JENKINS-42647 - error messages from github showing as json
- [x] ~https://issues.jenkins-ci.org/browse/JENKINS-42734 - empty branch name result in bad Jenkinsfile~ Fixed in https://github.com/jenkinsci/blueocean-plugin/pull/905
- [x] https://issues.jenkins-ci.org/browse/JENKINS-42649 - easter egg for load/save

### CODE REVIEW

**Comment by michaelneale** [comment]
@kzantow can the pom.xml point to rc1 of blue ocean - easier to test

**Comment by kzantow** [comment]
@michaelneale done, sorry forgot that :-/

**Comment by michaelneale** [comment]
:bee: nice one. 

one 🐜  is that the popup window for the easter egg is hyper narrow - but LGTM - can tweak that stuff as needed.

**Comment by michaelneale** [comment]
forgot:

File: src/main/js/EditorPage.jsx [hunk 1]
```diff
@@ -155,6 +162,39 @@ class PipelineLoader extends React.Component {
...
+        }
+
+        const showLoadingError = err => {
+            this.showErrorDialog(
+                <div className="errors">
+                    <div>
+                        There was an error loading the pipeline
+                        from the Jenkinsfile from your repository.
+                        Correct the error by editing the Jenkinsfile
+                        directly and committing it back to the repository.
```
Comment (i386) [inline src/main/js/EditorPage.jsx]: Small change:
```
There was an error loading the pipeline from the Jenkinsfile in this repository.
Correct the error by editing the Jenkinsfile using the declarative syntax then commit it back to the repository.
```

**Comment by michaelneale** [comment]
Tweak as above (James and I talked it over) and then :bee: get it in!

**Comment by michaelneale** [comment]
:bee: nice one - works well for me.

**Pull request merged**